### DISCUSSION TITLE
possible ttl bug in linux version of mgen

### DISCUSSION THREAD

**Discussion by PatrickTague** [USNavalResearchLaboratory/mgen#35] [General]
i'm trying to use mgen for traffic profiling in the CORE emulator.  however, for every TCP flow i try to create, it sets the TTL field to 3, regardless of what parameter i provide as part of the flow definition.  is there an easy way to fix this?

for context, i'm using the version of mgen that is installed via apt in ubuntu, i believe it's 5.02c.

**Comment by bebopagogo** [comment]
I think this has been addressed in the more recent source code that is on GitHub.  The intent was to have the _default_ TTL for _multicast_ to be 3 but at one point that also got mistakenly applied to unicast too.

**Reply by PatrickTague** [reply]
thank you for the weekend reply!  is there any way to change it to not be 3?  i'm unable to establish any longer connections, regardless of what configuration i try.

**Reply by bebopagogo** [reply]
Have you tried setting the global/default TTL on the command-line before any ON or LISTEN command?  For example:

mgen ttl 255 event "on 1 tcp dst 127.0.0.1/5002 per [10 1024]"

I tested this with a 5.02 build, observed the TTL=3 as you have (even with the TTL as part of the on event specification), but the above command line produced packets with TTL=255

**Reply by bebopagogo** [reply]
(I also tested this with the current MGEN 5..1.1 source code from GitHub and it uses TTL=255 for unicast flows.)

**Reply by PatrickTague** [reply]
ooh, that worked. thank you!  i was hoping to not need ~70 students in my class install from source if at all possible, so that will do it!
